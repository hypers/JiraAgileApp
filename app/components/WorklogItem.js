import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Image,
  Text,
  FlatList,
  TouchableHighlight,
  Platform,
  StyleSheet
} from 'react-native';
import _ from 'lodash';
import moment from 'moment';

import Avatar from './Avatar';

function formatSecond(time) {
  if (time % 3600 === 0) {
    return time / 3600 + ' 小时';
  } else {
    return (time / 3600).toFixed(1) + ' 小时';
  }
}

class WorklogItem extends PureComponent {
  render() {
    const { item, index, onLongPress } = this.props;
    return (
      <TouchableHighlight
        style={styles.card}
        /* onLongPress={(e) => onLongPress(e, item.id)} */
      >
        <View >
          <View style={styles.info}>
            <Avatar uri={item.author.avatarUrls['24x24']} width={24} height={24} style={styles.avatar} />
            <Text style={styles.authorName}>{item.author.displayName}</Text>
            <Text style={styles.time}>{moment(item.started).format('YYYY-MM-DD HH:mm')}</Text>
            <Text style={styles.timeSpend}>{formatSecond(item.timeSpentSeconds)}</Text>
          </View>
          {item.comment ? <Text style={styles.comment}>
            {item.comment}
          </Text> : null}
          
        </View>
      </TouchableHighlight>
    );
  }
}

export default WorklogItem;