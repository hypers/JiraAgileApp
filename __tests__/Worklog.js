import 'react-native';
import React from 'react';

import WorklogItem, { formatSecond } from '../app/components/WorklogItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const mockItem = {
  "self": "https://jira.hypers.com/rest/api/2/issue/28093/worklog/21632",
  "author": {
      "self": "https://jira.hypers.com/rest/api/2/user?username=jianbo.yu",
      "name": "jianbo.yu",
      "key": "jianbo.yu",
      "emailAddress": "jianbo.yu@hypers.com",
      "avatarUrls": {
          "48x48": "https://jira.hypers.com/secure/useravatar?ownerId=jianbo.yu&avatarId=11718",
          "24x24": "https://jira.hypers.com/secure/useravatar?size=small&ownerId=jianbo.yu&avatarId=11718",
          "16x16": "https://jira.hypers.com/secure/useravatar?size=xsmall&ownerId=jianbo.yu&avatarId=11718",
          "32x32": "https://jira.hypers.com/secure/useravatar?size=medium&ownerId=jianbo.yu&avatarId=11718"
      },
      "displayName": "喻建博",
      "active": true,
      "timeZone": "Asia/Shanghai"
  },
  "updateAuthor": {
      "self": "https://jira.hypers.com/rest/api/2/user?username=jianbo.yu",
      "name": "jianbo.yu",
      "key": "jianbo.yu",
      "emailAddress": "jianbo.yu@hypers.com",
      "avatarUrls": {
          "48x48": "https://jira.hypers.com/secure/useravatar?ownerId=jianbo.yu&avatarId=11718",
          "24x24": "https://jira.hypers.com/secure/useravatar?size=small&ownerId=jianbo.yu&avatarId=11718",
          "16x16": "https://jira.hypers.com/secure/useravatar?size=xsmall&ownerId=jianbo.yu&avatarId=11718",
          "32x32": "https://jira.hypers.com/secure/useravatar?size=medium&ownerId=jianbo.yu&avatarId=11718"
      },
      "displayName": "喻建博",
      "active": true,
      "timeZone": "Asia/Shanghai"
  },
  "comment": "",
  "created": "2017-10-24T18:50:01.000+0800",
  "updated": "2017-10-24T18:50:01.000+0800",
  "started": "2017-10-24T18:49:00.000+0800",
  "timeSpent": "4h",
  "timeSpentSeconds": 14400,
  "id": "21632",
  "issueId": "28093"
}

describe('Worklog', () => {

  test('formatSecond', () => {
    expect(formatSecond(3600)).toBe('1 小时');
    expect(formatSecond(1800)).toBe('0.5 小时');
    expect(formatSecond(360)).toBe('0.1 小时');
  });

  it('WorklogItem', () => {
    const tree = renderer.create(
      <WorklogItem
        item={mockItem}
      />
    );
    expect(tree).toMatchSnapshot();
  });

});

