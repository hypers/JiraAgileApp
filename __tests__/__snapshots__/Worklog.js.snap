// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Worklog WorklogItem 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://jira.hypers.com/secure/useravatar?size=small&ownerId=jianbo.yu&avatarId=11718",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 12,
              "height": 24,
              "marginTop": -2,
              "width": 24,
            },
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555",
            "marginLeft": 15,
          }
        }
      >
        喻建博
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555",
            "marginLeft": 10,
          }
        }
      >
        2017-10-24 18:49
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555",
            "position": "absolute",
            "right": 0,
          }
        }
      >
        4 小时
      </Text>
    </View>
  </View>
</View>
`;
